spec/requests/api/can_authenticate_based_on_user_role_and_client_source_spec.rb

```diff
@@ -0,0 +1,64 @@
+RSpec.describe 'POST /api/auth/sign_in' do
+  let!(:regular_user) { create(:user, role: 'member', email: 'ordinary_guy@random.com') }
+  let!(:journalist) { create(:user, role: 'journalist', email: 'pulizer_prize_guy@random.com') }
+
+  describe 'admin-system' do
+    describe 'as a member' do
+      before do
+        post '/api/auth/sign_in', params: {
+          email: 'ordinary_guy@random.com',
+          password: 'password',
+          source: 'admin-system'
+        }
+      end
+  
+      it do
+        expect(response).to have_http_status 401
+      end
+    end
+  
+    describe 'as a journalist' do
+      before do
+        post '/api/auth/sign_in', params: {
+          email: 'pulizer_prize_guy@random.com',
+          password: 'password',
+          source: 'admin-system'
+        }
+      end
+  
+      it do
+        expect(response).to have_http_status 200
+      end
+    end
+  end
+
+  describe 'public-system' do
```

Comment:
It may be out of scope, but I guess it is a corner case, so I'll let it fly this time.